pkg/machineoutput/types_event_logging.go

```diff
@@ -0,0 +1,86 @@
+package machineoutput
+
+import (
+	"io"
+)
+
+// MachineEventLoggingClient is an interface which is used by consuming code to output machine-readable
+// event JSON to the console. Both no-op and non-no-op implementations of this interface exist.
+type MachineEventLoggingClient interface {
+
+	// These functions output the corresponding eponymous JSON event to the console
+
+	DevFileCommandExecutionBegin(commandID string, componentName string, commandLine string, groupKind string, timestamp string)
```

Comment:
Nit: Could we rename this and `DevFileCommandExecutionComplete` to something else as they share the same name as the struct that they both return?

Reply:
I did this intentionally, but am open to better ideas: I considered prepending `Output` to each of the method names, but I liked the API simplicity of the name of the method corresponding to the JSON event that it emits.

Was there a specific use case you were concerned about, or just a general concern about it being non-idiomatic Go? One thing I would note is that the structs are never referenced outside of the `machineoutput` package (though I need still to capitalize them to allow the JSON (un)marshalling to work), so I don't think there is a concern about confusion over which is referred to.